applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -234,6 +242,299 @@ namespace MPMParticleGeneratorUtility
                 }
             }
         }
+    }
+    /**
+     * @brief Function to Initiate material point condition.
+     * @details Generating particle condition using a designated shape functions
+     */
+
+    template<SizeType TDimension>
+    void GenerateMaterialPointCondition(ModelPart& rBackgroundGridModelPart,
+                                            ModelPart& rInitialModelPart,
+                                            ModelPart& rMPMModelPart){
+        // Initialize zero the variables needed
+        std::vector<array_1d<double, 3>> mpc_xg = { ZeroVector(3) };
+        array_1d<double,3> mpc_normal = ZeroVector(3);
+        std::vector<array_1d<double, 3>> mpc_displacement = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_imposed_displacement = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_velocity = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_imposed_velocity = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_acceleration = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_imposed_acceleration = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> mpc_contact_force = { ZeroVector(3) };
+        std::vector<array_1d<double, 3>> point_load = { ZeroVector(3) };
+
+        std::vector<double> mpc_area(1);
+        std::vector<double> mpc_penalty_factor(1);
+
+        // Determine condition index: This convention is done in order for the purpose of visualization in GiD
+        const unsigned int number_conditions = rBackgroundGridModelPart.NumberOfConditions();
+        const unsigned int number_elements = rBackgroundGridModelPart.NumberOfElements() + rInitialModelPart.NumberOfElements() + rMPMModelPart.NumberOfElements();
+        const unsigned int number_nodes = rBackgroundGridModelPart.NumberOfNodes();
+        unsigned int last_condition_id;
+        if (number_elements > number_nodes && number_elements > number_conditions)
+            last_condition_id = number_elements + 1;
+        else if (number_nodes > number_elements && number_nodes > number_conditions)
+            last_condition_id = number_nodes + 1;
+        else
+            last_condition_id = number_conditions + 1;
+
+        BinBasedFastPointLocator<TDimension> SearchStructure(rBackgroundGridModelPart);
+        SearchStructure.UpdateSearchDatabase();
+        typename BinBasedFastPointLocator<TDimension>::ResultContainerType results(100);
+
+        // Loop over the submodelpart of rBackgroundGridModelPart
+        for (ModelPart::SubModelPartIterator submodelpart_it = rBackgroundGridModelPart.SubModelPartsBegin();
+                submodelpart_it != rBackgroundGridModelPart.SubModelPartsEnd(); submodelpart_it++)
```

Comment:
```suggestion
        for (auto& sub_model_part : rBackgroundGridModelPart.SubModelParts();)
```

Reply:
I added the suggested change to the generation of the elements and conditions